src/main/java/org/assertj/core/api/AbstractMapAssert.java

```diff
@@ -1136,7 +1136,10 @@ protected SELF doesNotContainKeysForProxy(K[] keys) {
 
   /**
    * Verifies that the actual map contains only the given keys and nothing else, in any order.
-   *
+   * <p>
+   * The verification tries to honor the key comparison semantic of the underlying map implementation, but there might
+   * be some cases where the semantic cannot be honored and not all the unexpected elements are identified correctly
+   * (e.g., due to immutability).
```

Comment:
What about:

> The verification tries to honor the key comparison semantic of the underlying map implementation. The map under test has to be cloned to identify unexpected elements, but depending on the map implementation this may not always be possible. In these cases, a regular map is used and the key comparison strategy may not be the same as the map under test.

Reply:
sounds good to me